controllers/runner_controller.go

```diff
@@ -874,30 +874,44 @@ func newRunnerPod(template corev1.Pod, runnerSpec v1alpha1.RunnerConfig, default
 	// When you're NOT using dindWithinRunner=true,
 	// it must also be shared with the dind container as it seems like required to run docker steps.
 	//
+	// Setting VolumeSizeLimit to zero will disable /runner emptydir mount
+	//
+	// Enabling VolumeUseMemory will use tmpfs (RAM-backed filesystem) for emptydirs
+	//
 
 	runnerVolumeName := "runner"
 	runnerVolumeMountPath := "/runner"
-	runnerVolumeEmptyDir := &corev1.EmptyDirVolumeSource{}
+	runnerVolumeEmptyDirMedium := corev1.StorageMediumDefault
+
+	if runnerSpec.VolumeUseMemory != nil && *runnerSpec.VolumeUseMemory {
+		runnerVolumeEmptyDirMedium = corev1.StorageMediumMemory
+	}
+
+	runnerVolumeEmptyDir := &corev1.EmptyDirVolumeSource{
+		Medium: runnerVolumeEmptyDirMedium,
+	}
 
 	if runnerSpec.VolumeSizeLimit != nil {
 		runnerVolumeEmptyDir.SizeLimit = runnerSpec.VolumeSizeLimit
 	}
 
-	pod.Spec.Volumes = append(pod.Spec.Volumes,
```

Comment:
Can we add some validation here so that this function returns an explicit error when you tried to disable the empty dir when `dockerdWithinRunnerContainer: false`?

The error might say something like `/runner volume can't be disabled because it is required to share the working directory between the runner and the dockerd containers`

Reply:
Done